tjpcov_new/covariance_gaussian_fsky.py

```diff
@@ -0,0 +1,263 @@
+
+        Parameters:
+        -----------
+        ang_min (int, float):
+            if do_xi, ang is assumed to be theta (arcmin)
+            if do_xi == False,  ang is assumed to be ell
+        Returns:
+        --------
+            (theta, theta_edges ) (degrees):
+        """
+        # FIXME:
+        # Use sacc is passing this
+        if not do_xi:
+            ang_delta = (ang_max-ang_min)//n_ang
+            ang_edges = np.arange(ang_min, ang_max+1, ang_delta)
+            ang = np.arange(ang_min, ang_max + ang_delta - 2)
+
+        if do_xi:
+            th_min = ang_min/60  # in degrees
+            th_max = ang_max/60
+            n_th_bins = n_ang
+            ang_edges = np.logspace(np.log10(th_min), np.log10(th_max),
+                                    n_th_bins+1)
+            th = np.logspace(np.log10(th_min*0.98), np.log10(1), n_th_bins*30)
+            # binned covariance can be sensitive to the th values. Make sure
+            # you check convergence for your application
+            th2 = np.linspace(1, th_max*1.02, n_th_bins*30)
+
+            ang = np.unique(np.sort(np.append(th, th2)))
+            ang_bins = 0.5 * (ang_edges[1:] + ang_edges[:-1])
+
+            return ang, ang_bins, ang_edges  # TODO FIXIT
+
+        return ang, ang_edges
+
+    def get_ell_theta(self, ang_scale='linear', do_xi=False):
+        """
+        Get ell or theta for bins given the sacc object
+        For now, presuming only log and linear bins
+
+        Parameters:
+        -----------
+
+        Returns:
+        --------
+        """
+        ang_name = "ell" if not do_xi else 'theta'
+
+        # assuming same ell for all bins:
+        sacc_file = self.get_sacc_file()
+        data_type = sacc_file.get_data_types()[0]
+        tracer_comb = sacc_file.get_tracer_combinations()[0]
+        ang_bins = sacc_file.get_tag(ang_name, data_type=data_type,
+                                     tracers=tracer_comb)
+
+        ang_bins = np.array(ang_bins)
+
+        angb_min, angb_max = ang_bins.min(), ang_bins.max()
+        if ang_name == 'theta':
+            # assuming log bins
+            del_ang = (ang_bins[1:]/ang_bins[:-1]).mean()
+            ang_scale = 'log'
+            assert 1 == 1
+
+        elif ang_name == 'ell':
+            # assuming linear bins
+            del_ang = (ang_bins[1:] - ang_bins[:-1])[0]
+            ang_scale = 'linear'
+
+        ang, ang_edges = self.set_ell_theta(angb_min-del_ang/2,
+                                            angb_max+del_ang/2,
+                                            len(ang_bins),
+                                            ang_scale=ang_scale, do_xi=do_xi)
+        # Sanity check
+        if ang_scale == 'linear':
+            assert np.allclose((ang_edges[1:]+ang_edges[:-1])/2, ang_bins), \
+                "differences in produced ell/theta"
+        return ang, ang_bins, ang_edges
+
+
+class RealGaussianFsky(CovarianceReal, FourierGaussianFsky):
```

Comment:
Double inheritance can sometimes lead to some tricky bugs (diamond problem), might make sense to see if this can be avoided, maybe in the future?

Reply:
Maybe we can instantiate the Fourier class inside the Real one when needed. I'll think more about it when I start with the test for the real space covariance